src/main/java/com/direwolf20/buildinggadgets/common/util/tools/SetBackedPlacementSequence.java

```diff
@@ -0,0 +1,91 @@
+package com.direwolf20.buildinggadgets.common.util.tools;
+
+import com.direwolf20.buildinggadgets.api.building.IPositionPlacementSequence;
+import com.direwolf20.buildinggadgets.api.building.Region;
+import com.direwolf20.buildinggadgets.common.util.helpers.NBTHelper;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.util.Constants;
+
+import javax.annotation.Nonnull;
+import java.io.*;
+import java.util.HashSet;
+import java.util.Iterator;
+
+public class SetBackedPlacementSequence implements IPositionPlacementSequence, Serializable {
+
+    private static final long serialVersionUID = 5752016672792062860L;
+
+    private Region boundingBox;
+    private HashSet<BlockPos> internalSet;
+
+    public SetBackedPlacementSequence(HashSet<BlockPos> internalSet, Region boundingBox) {
+        this.internalSet = internalSet;
+        this.boundingBox = boundingBox;
+    }
+
+    @Nonnull
+    @Override
+    public Iterator<BlockPos> iterator() {
+        return internalSet.iterator();
+    }
+
+    @Nonnull
+    @Override
+    public Region getBoundingBox() {
+        return boundingBox;
+    }
+
+    /**
+     * @deprecated Use {@link #contains(int, int, int)} instead.
+     */
+    @Deprecated
+    @Override
+    public boolean mayContain(int x, int y, int z) {
+        return contains(new BlockPos(x, y, z));
+    }
+
+    public boolean contains(int x, int y, int z) {
+        return contains(new BlockPos(x, y, z));
+    }
+
+    public boolean contains(BlockPos pos) {
+        return internalSet.contains(pos);
+    }
+
+    /**
+     * Warning: this method uses the copy constructor of {@link HashSet}, therefore it does not guarantee it will return
+     * the same type of set.
+     */
+    @Deprecated
```

Comment:
Is this part of our codebase or is this from external sources, if the code has not get been shipped we should really just remove this instead of Deprecating it. Just an opinion.

Reply:
@MichaelHillcox this is a Method from the interface IPlacementSequence